processing/src/main/java/org/apache/druid/frame/write/cast/ObjectToFloatColumnValueSelector.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.write.cast;
+
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.math.expr.ExprEval;
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import org.apache.druid.segment.ColumnValueSelector;
+import org.apache.druid.segment.RowIdSupplier;
+
+import javax.annotation.Nullable;
+
+public class ObjectToFloatColumnValueSelector implements ColumnValueSelector<Float>
+{
+  private final ColumnValueSelector<?> objectSelector;
+
+  @Nullable
+  private final RowIdSupplier rowIdSupplier;
+
+  @Nullable
+  private Float currentValue;
+  private long currentRowId = RowIdSupplier.INIT;
+
+  public ObjectToFloatColumnValueSelector(
+      final ColumnValueSelector<?> objectSelector,
+      @Nullable final RowIdSupplier rowIdSupplier
+  )
+  {
+    this.objectSelector = objectSelector;
+    this.rowIdSupplier = rowIdSupplier;
+  }
+
+  @Override
+  public double getDouble()
+  {
+    final Number n = computeIfNeeded();
+    return n == null ? NullHandling.ZERO_DOUBLE : n.floatValue();
```

Comment:
should this call n.doubleValue()?

Reply:
I figured since this is the "to float" selector, it should use float precision even if the caller is requesting a double.

Fwiw, the caller really shouldn't be requesting a double anyway (since it's a "to float" selector). I have tests for this method but I don't expect it to be called.